custom/enikshay/management/commands/drtb_2_to_2_dot_1_migration.py

```diff
@@ -0,0 +1,287 @@
+
+    def get_episode_case_status(self, episode_case, episode_cases):
+        episode_case_props = episode_case.dynamic_case_properties()
+        props_status = {'case_type': 'episode', 'case_id': episode_case.get_id}
+        for prop in self.episode_case_relevant_props:
+            props_status[prop] = episode_case_props.get(prop)
+        props_status['updates'] = self.episode_case_updates(episode_case, episode_cases)
+        return props_status
+
+    def get_test_case_status(self, test_case, recently_opened_episode_case, occurrence_case):
+        test_case_props = test_case.dynamic_case_properties()
+        props_status = {'case_type': 'test', 'case_id': test_case.get_id}
+        for prop in self.test_case_relevant_props:
+            props_status[prop] = test_case_props.get(prop)
+        recently_opened_episode_case, props_status['updates'] = self.test_case_updates(
+            test_case, recently_opened_episode_case, occurrence_case)
+        return props_status
+
+    @staticmethod
+    def person_case_updates(person_case):
+        # person.case_minor_version = 21
+        props_to_update = {
+            'case_minor_version': '21',
+        }
+
+        person_case_props = person_case.dynamic_case_properties()
+
+        # If person.aadhaar_number != '', save person.aadhaar_number_obfuscated which is
+        # person.aadhaar number masked with *s except the last 4 digits
+        aadhar_num = person_case_props.get('aadhaar_number')
+        if aadhar_num:
+            # mask everything except last 4 digits
+            aadhar_num_length = len(aadhar_num)
+            props_to_update['aadhaar_number_obfuscated'] = '*' * (aadhar_num_length - 4) + aadhar_num[-4:]
+
+        # if person.age is blank/null and person.age_entered != blank/null
+        # then set person.age = person.age_entered.
+        # else if person.age = '' and person.dob != '', person.age = int((today() - person.dob) / 365.25)
+        if not person_case_props.get('age') and person_case_props.get('age_entered'):
+            props_to_update['age'] = person_case_props.get('age_entered')
+        elif not person_case_props.get('age') and person_case_props.get('dob'):
+            props_to_update['age'] = int(
+                (date.today() - datetime.strptime(person_case_props.get('dob'), "%Y-%m-%d").date()).days / 365.25
+            )
+
+        # Set person.referred_outside_enikshay_by_id = person.referred_by_id if person.referred_by_id exists
+        if person_case_props.get('referred_by_id'):
+            props_to_update['referred_outside_enikshay_by_id'] = person_case_props.get('referred_by_id')
+
+        # if the owner id is not _archive_ set person.referred_outside_enikshay_date and
+        # person.referred_outside_enikshay_by_id to blank
+        if person_case.owner_id != ARCHIVED_CASE_OWNER_ID:
+            props_to_update['referred_outside_enikshay_date'] = ""
+            props_to_update['referred_outside_enikshay_by_id'] = ""
```

Comment:
@sheelio this seems kinda weird - `referred_outside_enikshay_by_id` is sometimes updated to reference another value, and also sometimes wiped?  Looks like it's implemented as per the instructions, just wondering if those instructions are correct.

Reply:
Yea, its a bit weird but this is correct. 

The first situation is just moving it from an old case property to a new case property (which happens for almost all referred outside of enikshay cases)

The second situation handles making sure its cleared / empty for the situation where you referred outside of enikshay and then it was brought back (by referring it back into enikshay).